frontend/components/user-settings.tsx

```diff
@@ -0,0 +1,388 @@
+                  <SelectTrigger id="skillLevel">
+                    <SelectValue placeholder="Select skill level" />
+                  </SelectTrigger>
+                  <SelectContent>
+                    <SelectItem value="Novice">Novice</SelectItem>
+                    <SelectItem value="Intermediate">Intermediate</SelectItem>
+                    <SelectItem value="Expert">Expert</SelectItem>
+                  </SelectContent>
+                </Select>
+              </div>
+            </CardContent>
+            <CardFooter>
+              <Button onClick={handleSaveChanges}>Save Changes</Button>
+            </CardFooter>
+          </Card>
+        </TabsContent>
+        <TabsContent value="account">
+          <Card>
+            <CardHeader>
+              <CardTitle>Account Security</CardTitle>
+              <CardDescription>Manage your password and account settings.</CardDescription>
+            </CardHeader>
+            <CardContent className="space-y-4">
+              <div>
+                <Label htmlFor="currentPassword">Current Password</Label>
+                <Input
+                  id="currentPassword"
+                  type="password"
+                  value={currentPassword}
+                  onChange={handlePasswordInputChange}
+                />
+              </div>
+              <div>
+                <Label htmlFor="newPassword">New Password</Label>
+                <Input
+                  id="newPassword"
+                  type="password"
+                  value={newPassword}
+                  onChange={handlePasswordInputChange}
+                />
+              </div>
+              <div>
+                <Label htmlFor="confirmPassword">Confirm New Password</Label>
+                <Input
+                  id="confirmPassword"
+                  type="password"
+                  value={confirmPassword}
+                  onChange={handlePasswordInputChange}
+                />
+              </div>
+            </CardContent>
+            <CardFooter>
+              <Button onClick={handleChangePassword}>Change Password</Button>
+            </CardFooter>
+          </Card>
+          <Card className="mt-4">
+            <CardHeader>
+              <CardTitle>Delete Account</CardTitle>
+              <CardDescription>Permanently delete your account and all associated data.</CardDescription>
+            </CardHeader>
+            <CardContent>
+              <Alert variant="destructive">
+                <AlertCircle className="h-4 w-4" />
+                <AlertTitle>Warning</AlertTitle>
+                <AlertDescription>
+                  This action cannot be undone. All your data will be permanently removed.
+                </AlertDescription>
+              </Alert>
+            </CardContent>
+            <CardFooter>
+              <Button variant="destructive">Delete Account</Button>
+            </CardFooter>
+          </Card>
```

Comment:
Could we abstract this into a DeleteAccount component? Similarly with the above to have a ForgetPassword component and etc

Reply:
Oh wait just realised this is from Selwyns, I think it will be better to point the PR on top of his instead of pointing towards main